wolfcrypt/src/ed448.c

```diff
@@ -179,37 +231,35 @@ static int ed448_sign_msg(const byte* in, word32 inLen, byte* out,
 
         /* step 1: create nonce to use where nonce is r in
            r = H(h_b, ... ,h_2b-1,M) */
-        ret = wc_Shake256Hash(key->k, ED448_KEY_SIZE, az, sizeof(az));
+        ret = ed448_hash(key, key->k, ED448_KEY_SIZE, az, sizeof(az));
     }
     if (ret == 0) {
         /* apply clamp */
         az[0]  &= 0xfc;
         az[55] |= 0x80;
         az[56]  = 0x00;
 
-        ret = wc_InitShake256(&sha, NULL, INVALID_DEVID);
         if (ret == 0) {
-            ret = wc_Shake256_Update(&sha, ed448Ctx, ED448CTX_SIZE);
+            ret = ed448_hash_update(key, ed448Ctx, ED448CTX_SIZE);
```

Comment:
Above `ret == 0` checked twice. Please remove 242 check, since it is done above.

Reply:
done